### PR TITLE
feat: Add ability to monitor progress and wait for run end for enterpriseStart

### PR DESCRIPTION
**Pull request by guilgaly**
Configuration example in the `build.gradle`:

```groovy
gatling {
    enterprise {
        waitForRunEnd true // or waitForRunEnd = true
    }
}
```

or run with e.g. `./gradlew gatlingEnterpriseStart -Dgatling.enterprise.waitForRunEnd=true`

### CODE REVIEW

**Review by slandelle — APPROVED** [review]

File: src/main/groovy/io/gatling/gradle/GatlingPluginExtension.groovy [hunk 1]
```diff
@@ -129,6 +131,14 @@ class GatlingPluginExtension {
             setSimulationClass(simulationClass)
         }
 
+        def setWaitForRunEnd(boolean waitForRunEnd) {
+            this.waitForRunEnd = waitForRunEnd
+        }
+
+        def waitForRunEnd(boolean waitForRunEnd) {
+            this.waitForRunEnd = waitForRunEnd
```
Comment (Isammoc) [inline src/main/groovy/io/gatling/gradle/GatlingPluginExtension.groovy:139]:
```suggestion
            setWaitForRunEnd(waitForRunEnd)
```

File: src/main/groovy/io/gatling/gradle/GatlingPluginExtension.groovy [hunk 2]
```diff
@@ -215,6 +225,17 @@ class GatlingPluginExtension {
...
         }
 
+        @Input
+        @Optional
+        boolean getWaitForRunEnd() {
+            if (waitForRunEnd) {
+                return waitForRunEnd
+            } else {
+                def systemWaitForRunEnd = System.getProperty(WAIT_FOR_RUN_END_PROPERTY)
+                return systemWaitForRunEnd ? Boolean.parseBoolean(systemWaitForRunEnd) : false
```
Comment (Isammoc) [inline src/main/groovy/io/gatling/gradle/GatlingPluginExtension.groovy:235]:
```suggestion
                return Boolean.parseBoolean(System.getProperty(WAIT_FOR_RUN_END_PROPERTY) ?: "false")
```

File: src/main/groovy/io/gatling/gradle/GatlingPluginExtension.groovy [hunk 3]
```diff
@@ -207,12 +217,17 @@ class GatlingPluginExtension {
...
+                ? batchMode
+                : Boolean.parseBoolean(System.getProperty(BATCH_MODE_PROPERTY))
+        }
+
+        @Input
+        @Optional
+        boolean getWaitForRunEnd() {
+            return waitForRunEnd
+                ? waitForRunEnd
+                : Boolean.parseBoolean(System.getProperty(WAIT_FOR_RUN_END_PROPERTY))
```
Comment (Isammoc) [inline src/main/groovy/io/gatling/gradle/GatlingPluginExtension.groovy:230]:
```suggestion
            return waitForRunEnd || Boolean.parseBoolean(System.getProperty(WAIT_FOR_RUN_END_PROPERTY))
```

File: src/main/groovy/io/gatling/gradle/GatlingPluginExtension.groovy [hunk 4]
```diff
@@ -207,12 +217,13 @@ class GatlingPluginExtension {
         @Input
         @Optional
         boolean getBatchMode() {
-            if (batchMode) {
-                return batchMode
-            } else {
-                def systemBatchMode = System.getProperty(BATCH_MODE_PROPERTY)
-                return systemBatchMode ? Boolean.parseBoolean(systemBatchMode) : false
-            }
+            return batchMode || Boolean.parseBoolean(System.getProperty(BATCH_MODE_PROPERTY))
```
Comment (slandelle) [inline src/main/groovy/io/gatling/gradle/GatlingPluginExtension.groovy:220]: Boolean.getBoolean(BATCH_MODE_PROPERTY)

@Isammoc @guilgaly noobs 😂

File: src/main/groovy/io/gatling/gradle/GatlingPluginExtension.groovy [hunk 5]
```diff
@@ -207,12 +217,13 @@ class GatlingPluginExtension {
...
+        }
+
+        @Input
+        @Optional
+        boolean getWaitForRunEnd() {
+            return waitForRunEnd || Boolean.parseBoolean(System.getProperty(WAIT_FOR_RUN_END_PROPERTY))
```
Comment (slandelle) [inline src/main/groovy/io/gatling/gradle/GatlingPluginExtension.groovy:226]: same as above

**Review by slandelle — CHANGES_REQUESTED** [review]

File: src/main/groovy/io/gatling/gradle/GatlingPluginExtension.groovy (hunk 3 above)
Comment (slandelle) [reply src/main/groovy/io/gatling/gradle/GatlingPluginExtension.groovy]: Nah. Boolean.getBoolean

File: src/main/groovy/io/gatling/gradle/GatlingPluginExtension.groovy (hunk 4 above)
Comment (guilgaly) [reply src/main/groovy/io/gatling/gradle/GatlingPluginExtension.groovy:220]: lol, I even already used that in the SBT plugin but not this one 😅

File: src/main/groovy/io/gatling/gradle/CommonLogMessage.groovy [hunk 6]
```diff
@@ -10,14 +10,29 @@ final class CommonLogMessage {
...
+                )
+            }
+            if (!waitForRunEndSetting) {
+                builder.append(
+                    """To wait for the end of the run when starting a simulation on Gatling Enterprise, add the following configuration:
+                      |gatling.enterprise.waitForRunEnd true
+                      |Or specify -Dgatling.enterprise.waitForRunEnd=true
+                      |
+                      |""".stripMargin())
+            }
```
Comment (guilgaly) [inline src/main/groovy/io/gatling/gradle/CommonLogMessage.groovy:33]: New log to notify the waitForRunEnd option is available

**Review by slandelle — APPROVED** [review]

**Review by Isammoc — APPROVED** [review]

**Pull request merged**